video/out/gl_video.c

```diff
@@ -1693,11 +1702,15 @@ static void pass_delinearize(struct gl_video *p, enum mp_csp_trc trc)
 }
 
 // Takes care of the main scaling and pre/post-conversions
-static void pass_scale_main(struct gl_video *p)
+static void pass_scale_main(struct gl_video *p, struct gl_transform offset)
 {
     // Figure out the main scaler.
     double xy[2];
     get_scale_factors(p, xy);
+    // Adjust the scale factors to account for the pre-scaler. For simplicity
+    // we just disregard shearing (no pre-scaler does shearing anyway)
+    xy[0] /= offset.m[0][0];
+    xy[1] /= offset.m[1][1];
```

Comment:
> Probably it's better to clear offset.t[] in that case, since scaler-resizes-only is explicitly stated by user anyway.

I would rather disable scale-resizes-only when using image doubling to be honest.